sky/clouds/kubernetes.py

```diff
@@ -919,3 +948,26 @@ def expand_infras(cls) -> List[str]:
             f'{cls.canonical_name()}/{c}'
             for c in cls.existing_allowed_contexts(silent=True)
         ]
+
+    @classmethod
+    def _cluster_supports_high_performance_networking(cls,
```

Comment:
we may need some caching mechanism for this to reduce the overhead for calling to the kubernetes API. Maybe add an lru cache for this function?

Reply:
Added